### PR TITLE
Lpd 26249 - refactor POSHI test

### PR DESCRIPTION
**Pull request by andrea-ale-sbarra**
https://liferay.atlassian.net/browse/LPD-26249.

OLD [CPCommerceProductOptions#CanAssertChangesNotSavedOnCorrespondingOptions](https://testray.liferay.com/home/-/testray/case_results/2570318665?redirect=https%3A%2F%2Ftestray.liferay.com%2Fhome%2F-%2Ftestray%2Fcase_results%2Findex%3Fstatuses%3D3%26testrayComponentId%3D1548301546%26testrayBuildId%3D2570318371%26cur%3D1%26delta%3D200%26orderByCol%3Dstatus_sortable%26orderByType%3Dasc%26status%3D3) made again with Playwright

OLD CPCommercePriceLists#AllowMultiplePriceEntriesInTheSamePriceList POSHI test made again with an integration test on API

OLD CPCommercePriceLists#AllowMultiplePriceEntriesInTheSamePromotion  POSHI test made again with an integration test on API

OLD CPCommerceProductOptions#CanSKUUpdateSuccessfullyWhenSelectingValuesFromMultipleOptions POSHI test made again with playwright

OLD CommerceProductDetails#CanViewSingleSkuVirtualProductDetailPage  POSHI test made again with playwright

### CODE REVIEW

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

File: modules/test/playwright/tests/commerce/commerce-product-content-web/productDetails.spec.ts [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+	await apiHelpers.headlessCommerceAdminCatalog.postSkuUnitOfMeasure(
+		sku1.id,
+		{
+			incrementalOrderQuantity: 3,
+			name: {en_US: 'Box'},
+			primary: true,
+			priority: 1,
+			rate: 1,
+		}
+	);
```
Comment (smottal) [inline modules/test/playwright/tests/commerce/commerce-product-content-web/productDetails.spec.ts:195]: can you move this near the sku1 stuff?

File: modules/test/playwright/tests/commerce/commerce-product-content-web/productDetails.spec.ts [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+		name: getRandomString(),
+		type: 'person',
+	});
+
+	apiHelpers.data.push({id: account.id, type: 'account'});
+
+	await apiHelpers.headlessAdminUser.assignUserToAccountByEmailAddress(
+		account.id,
+		['test@liferay.com']
+	);
```
Comment (smottal) [inline modules/test/playwright/tests/commerce/commerce-product-content-web/productDetails.spec.ts:217]: do you need the account?

**Review by smottal — CHANGES_REQUESTED** [review]

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/delivery/catalog/resource/v1_0/test/SkuResourceFunctionalTest.java [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+		BigDecimal commercePriceEntryPrice = commercePriceEntry.getPrice();
+
+		Assert.assertTrue(
+			Objects.equals(
+				price.getPromoPrice(), commercePriceEntryPrice.doubleValue()));
+	}
+
+	protected SkuResource skuResource;
+	protected Company testCompany;
+	protected Group testGroup;
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/delivery/catalog/resource/v1_0/test/SkuResourceFunctionalTest.java:240]: Why these are protected and not private?

File: modules/test/playwright/helpers/HeadlessCommerceAdminPricingApiHelper.ts [hunk 4]
```diff
@@ -42,6 +49,46 @@ export class HeadlessCommerceAdminPricingApiHelper {
...
+		);
+	}
+
+	async getBasePromoPriceListId(catalogId: number) {
+		return this.apiHelpers.get(
+			`${this.apiHelpers.baseUrl}${this.basePath}/price-lists?filter=catalogId/any(x:(x eq ${catalogId})) and catalogBasePriceList eq true and type eq 'promotion'&fields=id`
+		);
+	}
+
+	async postPriceEntry(priceEntry?: TPriceEntry) {
```
Comment (smottal) [inline modules/test/playwright/helpers/HeadlessCommerceAdminPricingApiHelper.ts:76]: is it correct that the priceEntry is optional?

File: modules/test/playwright/tests/commerce/commerce-product-content-web/productDetails.spec.ts [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+
+	await productDetailsPage.addProductDetailsWidget();
+
+	await page.goto(`/web/${site.name}/p/virtual`);
+
+	await expect(page.getByText('SkuVirtual')).toBeVisible();
+	await expect(page.getByText('mpn', {exact: true})).toBeVisible();
+	await expect(page.getByText('GTIN1')).toBeVisible();
+	await expect(page.getByText('Short description')).toBeVisible();
+
```
Comment (smottal) [inline modules/test/playwright/tests/commerce/commerce-product-content-web/productDetails.spec.ts:109]: please remove this empty line to have a single block of expect statements

File: modules/test/playwright/tests/commerce/commerce-product-content-web/productDetails.spec.ts [hunk 6]
```diff
@@ -0,0 +1,335 @@
...
+		/price-value-inactive/
+	);
+	await expect(page.getByText('$ 50.00')).toBeVisible();
+	await expect(page.getByText('$ 50.00')).toHaveClass(/price-value-promo/);
+	await expect(
+		page.getByText('Full description', {exact: true})
+	).toBeVisible();
+	await expect(
+		page.getByRole('link', {name: 'Download Sample File'})
+	).toBeVisible();
```
Comment (smottal) [inline modules/test/playwright/tests/commerce/commerce-product-content-web/productDetails.spec.ts:121]: these are all locators, should go in the page

File: modules/test/playwright/tests/commerce/commerce-product-content-web/productDetails.spec.ts [hunk 7]
```diff
@@ -0,0 +1,335 @@
...
+
+	await expect(page.getByLabel('Color')).toBeVisible();
+
+	await page.getByLabel('Color').selectOption({label: 'Black'});
+
+	await page.getByLabel('Size').selectOption({label: 'XL + $ 10.00'});
+
+	await expect(page.getByRole('cell', {name: 'Unit'})).toBeVisible();
+
+	await expect(page.getByText('$ 50.00')).toBeVisible();
```
Comment (smottal) [inline modules/test/playwright/tests/commerce/commerce-product-content-web/productDetails.spec.ts:334]: locators

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:test:rlevant

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:forward

**Comment by andrea-ale-sbarra** [comment]
ci:forward:force

**Pull request closed without merging**